src/app/compressed-air-assessment/centrifugal-graph/centrifugal-graph.component.ts

```diff
@@ -0,0 +1,263 @@
+    '#bcbd22',  // curry yellow-green
+    '#17becf'   // blue-teal
+  ];
+  plotlyMarkerShapes: Array<string> = [
+    'star',
+    'star-diamond',
+    'hexagram',
+    'star-square',
+    'square',
+    'diamond',
+    'cross',
+    'x',
+    'diamond-wide',
+    'diamond-tall'
+  ];
+
+  constructor(private inventoryService: InventoryService,
+    private compressedAirAssessmentService: CompressedAirAssessmentService,
+    private plotlyService: PlotlyService) { }
+
+  ngOnInit(): void {
+    if (!this.settings) {
+      this.settings = this.compressedAirAssessmentService.settings.getValue();
+    }
+    if (this.inReport) {
+      this.showAllCompressors = true;
+      this.compressedAirAssessment.compressorInventoryItems.forEach(compressor => {
+        if (compressor.nameplateData.compressorType == 6 && compressor.compressorControls.controlType != undefined) {
+          this.selectedCompressor = compressor;
+          this.centrifugalSpecifics = compressor.centrifugalSpecifics;
+          this.selectedCompressorType = compressor.nameplateData.compressorType;
+          this.showGraph = true;
+        } else {
+          this.showGraph = false;
+        }
+      });
+      this.drawChart();
+    } else {
+      this.dataSub = this.inventoryService.selectedCompressor.subscribe(val => {
+        this.selectedCompressor = val;
+        this.compressedAirAssessment = this.compressedAirAssessmentService.compressedAirAssessment.getValue();
+        this.centrifugalSpecifics = this.selectedCompressor.centrifugalSpecifics;
+        this.selectedCompressorType = this.selectedCompressor.nameplateData.compressorType;
+        if (this.selectedCompressorType == 6 && this.selectedCompressor.compressorControls.controlType != undefined) {
+          this.showGraph = true;
+        } else {
+          this.showGraph = false;
+        }
+        this.drawChart();
+      });
+    }
+  }
+
+  ngOnDestroy() {
+    if (!this.inReport) {
+      this.dataSub.unsubscribe();
+    }
+  }
+
+  ngOnChanges() {
+    if (this.centrifugalGraph) {
+      this.drawChart();
+    }
+  }
+
+  ngAfterViewInit() {
+    this.drawChart();
+    window.dispatchEvent(new Event("resize"));
+  }
+
+  drawChart() {
+    if (this.printView) {
+      this.showAllCompressors = true;
+    }
+    if (this.centrifugalGraph && this.compressedAirAssessment) {
+      let traceData: Array<TraceData> = new Array();
+      if (this.showGraph) {
+        if (this.showAllCompressors) {
+          let currentColorIndex: number = -1;
+          this.compressedAirAssessment.compressorInventoryItems.forEach(compressor => {
+            let currentTraceColor: string;
+            if (compressor.nameplateData.compressorType == 6 && compressor.compressorControls.controlType != undefined) {
+              if (currentColorIndex == 9) {
+                currentColorIndex = -1;
+              }
+              currentColorIndex++;
+              currentTraceColor = this.plotlyTraceColors[currentColorIndex];
```

Comment:
There is some boilerplate logic/object properties in the trace data creation. Create a method that take x/y array and color, etc and returns trace data